pulsar/src/utils.rs

```diff
@@ -388,3 +386,38 @@ impl<'de> Deserialize<'de> for Rewards {
         Ok(Rewards(value))
     }
 }
+
+/// move data from old directory (if any) to new directory, or else just create
+/// the new directory.
+pub(crate) fn create_or_move_data(old_dir: &PathBuf, new_dir: &PathBuf) -> Result<()> {
+    if old_dir == new_dir {
+        bail!("This directory is already set");
+    }
+
+    #[cfg(any(target_os = "macos", target_os = "linux"))]
+    if !new_dir.starts_with("/") {
+        bail!("New directory path must start with /");
+    }
+
+    // Create the new directory if it doesn't exist
+    if !new_dir.exists() {
+        fs::create_dir_all(new_dir)?;
+    }
+
+    // Move contents if the old directory exists
+    if old_dir.exists() {
+        let entries = fs::read_dir(old_dir)?;
+        for entry in entries {
+            let entry = entry?;
+            let file_name = entry.file_name();
+            fs::rename(entry.path(), new_dir.join(file_name))?;
```

Comment:
What would happen if this fails? We are left with some data in old directory and some data in new one. Which means after this happen, user won't be able to use pulsar and have to fix the situation by hand.

I would like to have atomic move implemented here. Either we move everything or we move nothing.